test/integration_tests/prototype/hifi_gan_pipeline_test.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+import torch
+import torchaudio
+import torchaudio.functional as F
+from torchaudio.prototype.pipelines import HIFIGAN_VOCODER_V3_LJSPEECH
+
+
+def test_smoke_hifi_gan_bundle():
+    """Smoke test of downloading weights for pretraining model"""
+    HIFIGAN_VOCODER_V3_LJSPEECH.get_vocoder()
+
+
+@pytest.mark.parametrize(
+    "lang",
+    [
+        ("en"),
+        ("en2"),
+    ],
+)
+def test_hifi_gan_pretrained_weights(lang, sample_speech):
+    """Test that a waveform reconstructed from mel spectrogram by HiFiGAN bundle is close enough to the original"""
+    bundle = HIFIGAN_VOCODER_V3_LJSPEECH
+
+    # Get HiFiGAN-compatible transformation from waveform to mel spectrogram
+    mel_transform = bundle.get_mel_transform()
+    # Get HiFiGAN vocoder
+    vocoder = bundle.get_vocoder()
+
+    ref_waveform, ref_sample_rate = torchaudio.load(sample_speech)
+    ref_waveform = F.resample(ref_waveform, orig_freq=ref_sample_rate, new_freq=bundle.sample_rate)
+    ref_waveform = ref_waveform[:, : -(ref_waveform.shape[1] % mel_transform.hop_size)]
```

Comment:
Looks like the test logic does not concern if the data used should be real speech or not. Can we replace this with some pseudo data? Since it's about (mel-)spectrogram,I think a sine sweep that covers all the frequency range bellow Nyquist frequency is well suited here.

something like [this](https://pytorch.org/audio/main/tutorials/oscillator_tutorial.html#changing-frequencies-across-time)

Reply:
Sure, I replaced the real audio with synthetic one which changes its frequency from 0 to Nyquist frequency.

The threshold on relative RME had to be raised from 2% to 4% though. This is probably because HiFiGAN was trained on mel spectrograms of real audio clips (English speech), and might work worse on artificial waveforms like the one I'm using.